instrumentation/httpurlconnection/README.md

```diff
@@ -0,0 +1,75 @@
+# Android Instrumentation for URLConnection, HttpURLConnection and HttpsURLConnection
+
+## Status : Experimental
+
+Provides OpenTelemetry instrumentation for:
+- [URLConnection](https://developer.android.com/reference/java/net/URLConnection)
+- [HttpURLConnection](https://developer.android.com/reference/java/net/HttpURLConnection)
+- [HttpsURLConnection](https://developer.android.com/reference/javax/net/ssl/HttpsURLConnection)
+
+## Quickstart
+
+### Overview
+
+This plugin enhances the Android application host code by instrumenting all critical APIs (specifically those that initiate a connection). It intercepts calls to these APIs to ensure the following actions are performed:
+- Context is added for distributed tracing before actual API is called (i.e before connection is initiated).
+- Traces and spans are generated and properly closed.
+- Any exceptions thrown are recorded.
+
+A span associated with a given request is concluded in the following scenarios:
+- When the getInputStream()/getErrorStream() APIs are called, the span concludes after the stream is fully read, an IOException is encountered, or the stream is closed.
+- When the disconnect API is called.
+
+Spans won't be automatically ended and reported otherwise. If any of your URLConnection requests do not call the span concluding APIs mentioned above, refer the section entitled ["Scheduling Harvester Thread"](#scheduling-harvester-thread). This section provides guidance on setting up a recurring thread that identifies unreported, idle connections (those that have been read from but have been inactive for more than 10 seconds) and concludes any open spans associated with them.
+
+> The minimum supported Android SDK version is 21, though it will also instrument APIs added in the Android SDK version 24 when running on devices with API level 24 and above.
+
+> If your project's minSdk is lower than 26, then you must enable
```

Comment:
Maybe we can pull out the similar test mentioned in main readme to a separate section in main readme itself,  something like "Critical Notes"(there is a slight more addition to that we can make - for AGP < 8.3.0, the property to be used for correct dexing is different). And link to that section from this readme. As this module can be used separately to the main module, it might be important to have that information linked from this readme as well. Does that sound good to you as well?

Reply:
I did a similar duplication in the Java sem conv repo earlier but was advised to put the details in one place so if we need to update, we didn't have to do it multiple places. But I think both ways have advantages, so it's up to you which one you prefer. I think merging as is and figuring it out later is perfectly reasonable.